pkg/cloudprovider/aws/fleet/launchtemplate.go

```diff
@@ -142,27 +142,33 @@ func (p *LaunchTemplateProvider) getSecurityGroupIds(ctx context.Context, cluste
 	return securityGroupIds, nil
 }
 
-func (p *LaunchTemplateProvider) getAMIID(context context.Context) (*string, error) {
+func (p *LaunchTemplateProvider) getAMIID(ctx context.Context) (*string, error) {
 	version, err := p.kubeServerVersion()
 	if err != nil {
 		return nil, fmt.Errorf("kube server version, %w", err)
 	}
-	paramOutput, err := p.ssm.GetParameter(&ssm.GetParameterInput{
-		Name: aws.String(fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended", version)),
+	paramOutput, err := p.ssm.GetParameterWithContext(ctx, &ssm.GetParameterInput{
+		Name: aws.String(fmt.Sprintf("/aws/service/bottlerocket/aws-k8s-%s/x86_64/latest/image_id", version)),
 	})
 	if err != nil {
-		return nil, fmt.Errorf("ssm get parameter, %w", err)
+		return nil, fmt.Errorf("getting ssm parameter, %w", err)
 	}
-	output := struct {
-		ImageID string `json:"image_id"`
-	}{}
-	if err := json.Unmarshal([]byte(*paramOutput.Parameter.Value), &output); err != nil {
-		return nil, fmt.Errorf("unmarshal parameter output, %w", err)
+	return paramOutput.Parameter.Value, nil
+}
+
+func (p *LaunchTemplateProvider) getUserData(cluster *v1alpha1.ClusterSpec) (*string, error) {
+	t := template.Must(template.New("userData").Parse(bottlerocketUserData))
+	var userData bytes.Buffer
+	if err := t.Execute(&userData, cluster); err != nil {
+		return nil, err
 	}
-	return &output.ImageID, nil
+	return aws.String(base64.StdEncoding.EncodeToString(userData.Bytes())), nil
 }
 
 func (p *LaunchTemplateProvider) kubeServerVersion() (string, error) {
 	version, err := p.clientSet.Discovery().ServerVersion()
+	if err != nil {
+		return "", err
+	}
 	return fmt.Sprintf("%s.%s", version.Major, strings.TrimSuffix(version.Minor, "+")), err
```

Comment:
nit: idiomatically, this should be 
```
	return fmt.Sprintf("%s.%s", version.Major, strings.TrimSuffix(version.Minor, "+")), nil
```
err -> nil

Reply:
doesn't that crash if `version` is `nil`?

Reply:
Whoops! Sorry I misread that!